File: SurveyAPI/SurveyAPI/Models/QuestionFull.cs
﻿namespace SurveyAPI.Models {
    public class QuestionFull {
        public Guid Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public DateTime BeginDate { get; set; }
        public DateTime EndDate { get; set; }
        public bool Multiple { get; set; }
        public List<ChoiceFull> Choices { get; set; } = new();
        public List<Guid> UserVotes { get; set; } = new();
    }
}

File: front-vue/src/models/question/QuestionFull.ts
import { ChoiceFullDTO } from "./ChoiceFull";

export class QuestionFullDTO {
  id = "";
  name = "";
  beginDate = "";
  endDate = "";
  multiple = false;
  choices: ChoiceFullDTO[] = [];
  userVotes: string[] = [];
}

export class QuestionFull extends QuestionFullDTO {
  constructor(data: QuestionFull | QuestionFullDTO | null) {
    super();
    if (data) Object.assign(this, data);
  }
}
